homeassistant/components/frontend/__init__.py

```diff
@@ -541,6 +546,23 @@ def get(self, request):
         })
 
 
+class TranslationsView(HomeAssistantView):
+    """View to return backend defined translations."""
+
+    url = '/api/translations/{language}'
+    name = 'api:translations'
+
+    @callback
+    def get(self, request, language):
+        """Return themes."""
+        hass = request.app['hass']
+
+        resources = hass.data[DATA_TRANSLATIONS].get(language)
```

Comment:
I would prefer if we structure this like we just restructured service descriptions: on demand.

So have this view look at which components are loaded (`hass.config.components`), look at their file names and see if they have files. For example, the light folder would look like:
 - `__init__.py`
 - `services.yaml`
 - `strings.json`
 - `strings.nl.json`

Bonus points if the frontend would be able to fetch the strings on a per component basis. So if it sees a string for `component.light.setup.bla`, it will only fetch the light strings from the backend.